<launch>
  <arg name="num_robots" value="5" />
  <arg name="robust_cost_type" default="L2" />
  <arg name="acceleration" default="true"/>
  <arg name="max_iteration_number" default="1000"/>

  <!-- Launch ROS node to publish pose graph -->
  <param name="/num_robots" type="int" value="$(arg num_robots)" />
  <param name="/dataset" type="str" value="$(find dpgo_ros)/data/sphere2500.g2o" />
  <node name="dataset_publisher" pkg="dpgo_ros" type="dpgo_ros_dataset_publisher_node" output="screen"/>

  <!-- Launch indivial PGO agents. Number must match /num_robots -->
  <group ns="kimera0">
    <include file="$(find dpgo_ros)/launch/PGOAgent.launch" pass_all_args="true">
      <arg name="agent_id" value="0" />
      <arg name="robust_cost_type" value="$(arg robust_cost_type)" />
      <arg name="acceleration" value="$(arg acceleration)" />
      <arg name="max_iteration_number" value="$(arg max_iteration_number)"/>
    </include>
  </group>

  <group ns="kimera1">
    <include file="$(find dpgo_ros)/launch/PGOAgent.launch" pass_all_args="true">
      <arg name="agent_id" value="1" />
      <arg name="robust_cost_type" value="$(arg robust_cost_type)" />
      <arg name="acceleration" value="$(arg acceleration)" />
      <arg name="max_iteration_number" value="$(arg max_iteration_number)"/>
    </include>
  </group>

  <group ns="kimera2">
    <include file="$(find dpgo_ros)/launch/PGOAgent.launch" pass_all_args="true">
      <arg name="agent_id" value="2" />
      <arg name="robust_cost_type" value="$(arg robust_cost_type)" />
      <arg name="acceleration" value="$(arg acceleration)" />
      <arg name="max_iteration_number" value="$(arg max_iteration_number)"/>
    </include>
  </group>

  <group ns="kimera3">
    <include file="$(find dpgo_ros)/launch/PGOAgent.launch" pass_all_args="true">
      <arg name="agent_id" value="3" />
      <arg name="robust_cost_type" value="$(arg robust_cost_type)" />
      <arg name="acceleration" value="$(arg acceleration)" />
      <arg name="max_iteration_number" value="$(arg max_iteration_number)"/>
    </include>
  </group>

  <group ns="kimera4">
    <include file="$(find dpgo_ros)/launch/PGOAgent.launch" pass_all_args="true">
      <arg name="agent_id" value="4" />
      <arg name="robust_cost_type" value="$(arg robust_cost_type)" />
      <arg name="acceleration" value="$(arg acceleration)" />
      <arg name="max_iteration_number" value="$(arg max_iteration_number)"/>
    </include>
  </group>

  <!-- Launch visualization in rviz -->
  <node name="rviz" pkg="rviz" type="rviz" output="screen" args="-d $(find dpgo_ros)/rviz/default.rviz"/>

</launch>